packages/xo-web/src/xo-app/index.js

```diff
@@ -114,6 +116,30 @@ const BODY_STYLE = {
     signedUp: !!state.user,
   }
 })
+@provideState({
+  initialState: () => ({ checkXoaCount: 0 }),
+  effects: {
+    async forceRefreshXoaStatus() {
+      await clearXoaCheckCache()
+      await this.effects.refreshXoaStatus()
+    },
+    refreshXoaStatus() {
+      this.state.checkXoaCount += 1
+    },
+  },
+  computed: {
+    xoaStatus: {
+      get({ checkXoaCount }) {
+        // To avoid aggressive minification which would remove destructuration
+        noop(checkXoaCount)
+        return getXoaPlan() === 'Community' ? '' : checkXoa().catch(() => 'XOA plugin not loaded')
+      },
+      placeholder: '',
+    },
+    isXoaStatusOk: ({ xoaStatus }) => !xoaStatus.includes('✖'),
+    xoaRegistered: ({ xoaStatus }) => xoaStatus !== 'XOA plugin not loaded',
```

Comment:
Why do you use this specific test to know if the XOA is registered?

Reply:
I didn't saw we had a method `getPlugin(id)`